make warnings php7.2.3

```
libtool: compile:  cc -g -O2 -O2 -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/tmp/igbinary-2.0.5 -DPHP_ATOM_INC -I/tmp/igbinary-2.0.5/include -I/tmp/igbinary-2.0.5/main -I/tmp/igbinary-2.0.5 -I/usr/include/php/20170718 -I/usr/include/php/20170718/main -I/usr/include/php/20170718/TSRM -I/usr/include/php/20170718/Zend -I/usr/include/php/20170718/ext -I/usr/include/php/20170718/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/igbinary-2.0.5/src/php7/igbinary.c  -fPIC -DPIC -o src/php7/.libs/igbinary.o
In file included from /usr/include/php/20170718/Zend/zend_modules.h:26:0,
                 from /usr/include/php/20170718/Zend/zend_API.h:27,
                 from /usr/include/php/20170718/main/php.h:40,
                 from /tmp/igbinary-2.0.5/src/php7/igbinary.c:18:
/usr/include/php/20170718/Zend/zend_compile.h:776:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
 ZEND_API void zend_try_exception_handler();
 ^
In file included from /tmp/igbinary-2.0.5/src/php7/igbinary.c:18:0:
/usr/include/php/20170718/main/php.h:373:46: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
 static inline ZEND_ATTRIBUTE_DEPRECATED void php_std_error_handling() {}
                                              ^
In file included from /usr/include/php/20170718/main/php.h:40:0,
                 from /tmp/igbinary-2.0.5/src/php7/igbinary.c:18:
/usr/include/php/20170718/Zend/zend_API.h:152:35: warning: no previous prototype for ‘get_module’ [-Wmissing-prototypes]
  ZEND_DLEXPORT zend_module_entry *get_module(void) { return &name##_module_entry; }\
                                   ^
/tmp/igbinary-2.0.5/src/php7/igbinary.c:305:1: note: in expansion of macro ‘ZEND_GET_MODULE’
 ZEND_GET_MODULE(igbinary)
 ^
/tmp/igbinary-2.0.5/src/php7/igbinary.c: In function ‘php_igbinary_init_globals’:
/tmp/igbinary-2.0.5/src/php7/igbinary.c:316:62: warning: declaration of ‘igbinary_globals’ shadows a global declaration [-Wshadow]
 static void php_igbinary_init_globals(zend_igbinary_globals *igbinary_globals) {
                                                              ^
In file included from /usr/include/php/20170718/main/php.h:40:0,
                 from /tmp/igbinary-2.0.5/src/php7/igbinary.c:18:
/tmp/igbinary-2.0.5/src/php7/igbinary.c:301:29: note: shadowed declaration is here
 ZEND_DECLARE_MODULE_GLOBALS(igbinary)
                             ^
/usr/include/php/20170718/Zend/zend_API.h:178:31: note: in definition of macro ‘ZEND_DECLARE_MODULE_GLOBALS’
  zend_##module_name##_globals module_name##_globals;
                               ^
/bin/bash /tmp/igbinary-2.0.5/libtool --mode=compile cc -g -O2 -O2 -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/tmp/igbinary-2.0.5 -DPHP_ATOM_INC -I/tmp/igbinary-2.0.5/include -I/tmp/igbinary-2.0.5/main -I/tmp/igbinary-2.0.5 -I/usr/include/php/20170718 -I/usr/include/php/20170718/main -I/usr/include/php/20170718/TSRM -I/usr/include/php/20170718/Zend -I/usr/include/php/20170718/ext -I/usr/include/php/20170718/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/igbinary-2.0.5/src/php7/hash_si.c -o src/php7/hash_si.lo 
```
Php version:
```
PHP 7.2.3-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Mar  6 2018 11:18:25) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.3-1+ubuntu16.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,6 +15,8 @@ php:
 # For unsupported versions(5.4-5.5), run minimal tests to cover 32-bit, 64-bit, and clang
 # Exclude everything else.
 matrix:
+  allow_failures:
+  - php: master
   exclude:
   - php: 5.4
     env: CC=clang   CFLAGS="-g -O0"

--- a/config.m4
+++ b/config.m4
@@ -67,7 +67,7 @@ if test "$PHP_IGBINARY" != "no"; then
   elif test "$GCC" = yes; then
     AC_MSG_RESULT(gcc)
     if test -z "`echo $CFLAGS | grep -- '-O[0123]'`"; then
-      PHP_IGBINARY_CFLAGS="$CFLAGS -O2 -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000"
+      PHP_IGBINARY_CFLAGS="$CFLAGS -O2 -Wall -Wpointer-arith -Wcast-align -Wwrite-strings -Wswitch -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000"
     fi
   elif test "$ICC" = yes; then
     AC_MSG_RESULT(icc)

